src/Forms/_FormValidationOverlay.test.js

```diff
@@ -0,0 +1,137 @@
+        document.body.innerHTML = '';
+    });
+
+    const getFormMessage = () => document.body.querySelector('.fd-popover__popper > div > .fd-form-message');
+
+    describe('default rendering', () => {
+        test('should render', () => {
+            let element = setup();
+            expect(element).toBeDefined();
+        });
+
+        test('should not render a FormMessage by default', () => {
+            setup();
+
+            const messageNode = getFormMessage();
+
+            expect(messageNode).toBeNull();
+        });
+    });
+
+    describe('Props', () => {
+        test('should allow props to be spread to the wrapping div component', () => {
+            let element = setup({ wrapperProps: { 'data-sample': 'Sample' } });
+
+            expect(
+                element.getDOMNode().attributes['data-sample'].value
+            ).toBe('Sample');
+        });
+
+        test('should set className on the outer div', () => {
+            const wrapper = setup({
+                className: 'wonderful-styles'
+            });
+
+            expect(
+                wrapper.find('.fd-popover').getDOMNode().classList
+            ).toContain('wonderful-styles');
+        });
+
+        test('should set className on the reference component', () => {
+            const wrapper = setup({
+                referenceClassName: 'wonderful-styles'
+            });
+
+            expect(
+                wrapper.find('.fd-popover__control').getDOMNode().classList
+            ).toContain('wonderful-styles');
+        });
+
+        test('should allow props to be spread to the FormMessage component', async() => {
+            await act(async() => {
+                setup({
+                    formMessageProps: { 'data-sample': 'Sample' },
+                    validationState: { state: 'error', message: 'test message' },
+                    show: true
+                });
+            });
+
+            const messageNode = getFormMessage();
+
+            expect(
+                messageNode.attributes['data-sample'].value
+            ).toBe('Sample');
+        });
+
+        test('should not show the FormMessage if there is no text', async() => {
+            const element = setup({
+                validationState: { state: 'error', message: 'test message' }
+            });
+
+            await act(async() => {
+                element.find('.fd-popover').simulate('click');
+            });
+
+            const messageNode = getFormMessage();
+
+            expect(messageNode).toBeNull();
+        });
+
+    });
+
+    describe('interaction', () => {
+        test('should show the FormMessage onFocus', async() => {
+            let element = setup({ validationState: { state: 'error', message: 'test message' } });
+
+            await act(async() => {
+                element.find('.fd-popover').simulate('focus');
+            });
+
+            const messageNode = getFormMessage();
+
+            expect(messageNode).toBeDefined();
```

Comment:
I'm pretty sure `expect(null).toBeDefined()` passes.  I f`querySelector` can't find an element it returns `null`, so this expect will always pass.

Reply:
Well... crap haha bad unit test!

Reply:
The one after this is going to have the same problem - I'll rework both